lib/schematic/stages/AddLibrarySymbolsStage.ts

```diff
@@ -319,12 +320,17 @@ export class AddLibrarySymbolsStage extends ConverterStage<
   private createPolylineFromPoints(
     points: Array<{ x: number; y: number }>,
     scale: number,
+    center: { x: number; y: number } | undefined,
     fillType: "none" | "background" = "none",
   ): SymbolPolyline {
     const polyline = new SymbolPolyline()
 
     // Scale points to match the c2kMatSch transformation scale
-    const xyPoints = points.map((p) => new Xy(p.x * scale, p.y * scale))
+    const cx = center?.x ?? 0
+    const cy = center?.y ?? 0
+    const xyPoints = points.map(
+      (p) => new Xy((p.x - cx) * scale, (p.y - cy) * scale),
```

Comment:
use transformation-matrix to perform any operations on points, e.g. the `applyToPoint` method. Never do point math with multiplication without a transformation-matrix